src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -127,8 +127,9 @@ public void Initialize(Http3StreamContext context)
             _frameWriter.Reset(context.Transport.Output, context.ConnectionId);
         }
 
-        public void InitializeWithExistingContext(IDuplexPipe transport)
+        public void InitializeWithExistingContext(IDuplexPipe transport, string connectionId)
```

Comment:
Should this just take a Http3StreamContext?

Reply:
The Http3StreamContext would need to be made public on Http3Stream to be able to pass it back in again.

The current code follows what Http2Stream does. The difference is there are a couple of properties that need to be updated on the context.